src/RpcClient.ts

```diff
@@ -354,4 +372,119 @@ export class RpcClient {
       throw new Error(`Error in getSignaturesForAsset: ${error}`);
     }
   }
+  /**
+   * Mint a compressed NFT.
+   * @returns {Promise<MintCNFTResponse>}
+   * @throws {Error}
+   */
+  async mintCNFT(
+    request: MintCNFTRequest
+  ): Promise<MintCNFTResponse | undefined> {
+    try {
+      const ownerKeypair = request.ownerKeypair;
+      // User can override defaulted values.
+      const defaultMetadataArgs: RequiredMetadataArgs = {
+        primarySaleHappened: false,
+        isMutable: false,
+        editionNonce: 0,
+        tokenStandard: null,
+        uses: null,
+        tokenProgramVersion: null,
+        ...request.metadataArgs,
+      };
+      let assetId = "";
+
+      let treeWallet = request.treeKeypair ?? Keypair.generate();
+      if (!request.treeKeypair) {
+        await initTree(this.connection, ownerKeypair, treeWallet);
+      }
+      let response: any;
+      let collection = request.collection ?? false;
+
+      // User set collection to false, or it was not dictated.
+      if (collection == false && treeWallet) {
```

Comment:
I'm wondering what purpose the ` && treeWallet` check serves. Do we have a scenario where `treeWallet` could be null? We are doing `let treeWallet = request.treeKeypair ?? Keypair.generate()` before coming here.